.github/workflows/deploy-artifact.yml

```diff
@@ -83,3 +83,65 @@ jobs:
           asset_path: ${{ steps.release.outputs.artifacts_archive_path }}
           asset_name: ${{ steps.release.outputs.artifacts_archive_path }}
           asset_content_type: application/zip
+
+  deploy-to-docker:
+    runs-on: ubuntu-latest
+    needs: [test]
+
+    env:
+      OWNER: "camunda"
+      IMAGE_NAME: "zeebe-process-test-engine"
+      TAG: "latest"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v2.4.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/common/github.com/actions/camunda-cloud/zeebe-process-test REGISTRY_HUB_DOCKER_COM_USR;
+            secret/data/common/github.com/actions/camunda-cloud/zeebe-process-test REGISTRY_HUB_DOCKER_COM_PSW;
+
+      - name: Set up Java environment
+        uses: actions/setup-java@v2.5.0
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Build jar
+        run: |
+          cd engine
+          mvn clean package -DskipTests
+
+      - name: Build Docker image
+        run: |
+          if ! echo $TAG | grep SNAPSHOT; then
```

Comment:
Interesting way to check the value

❓ is this a common way to check this?

If not, you might want to change it to `if [ "$TAG" != "SNAPSHOT" ]; then`, which I believe is the common way to check inequality of 2 strings in a shell script.

Reply:
Your solution does something different than my solution does. Mine checks if `$TAG` contains the text `SNAPSHOT`. Yours checks if `$TAG` is not equal to `SNAPSHOT`. Generally `$TAG` will be filled with something like `1.0-SNAPSHOT` which means we don't want to enter the if-statement.

Reply:
Interesting. I hadn't thought of that. Thanks for sharing